gap/json.gi

```diff
@@ -50,47 +50,53 @@ InstallMethod(_GapToJsonStreamInternal, [IsOutputStream, IsString],
 function(o, s)
   if IsEmpty(s) then
     if IsStringRep(s) then
-      PrintTo(o, "\"\"");
+      WriteAll(o, "\"\"");
     else
-      PrintTo(o, "[]");
+      WriteAll(o, "[]");
     fi;
   else
-    PrintTo(o, "\"", JSON_ESCAPE_STRING(s), "\"");
+    WriteAll(o, "\"");
+    WriteAll(o, JSON_ESCAPE_STRING(s));
+    WriteAll(o, "\"");
   fi;
 end );
 
 InstallMethod(_GapToJsonStreamInternal, [IsOutputStream, IsList],
 function(o, l)
   local i, first;
-  first := true;
-  PrintTo(o, "[");
-  for i in l do
-    if first then
-      first := false;
-    else
-      PrintTo(o, ",");
-    fi;
-    _GapToJsonStreamInternal(o, i);
-  od;
-  PrintTo(o, "]");
+  if IsOutputTextStream(o) and IsStringRep(o![1]) and IsPlistRep(l) then
+    GAP_PLIST_TO_JSON_STRING(o![1], o, l);
+  else
+    first := true;
+    WriteAll(o, "[");
```

Comment:
According to the Codecov comments, this part of the code is not covered by the test suite.